darts/timeseries.py

```diff
@@ -3607,6 +3607,20 @@ def _get_kwargs(transformation, forecasting_safe):
         if isinstance(transforms, dict):
             transforms = [transforms]
 
+        comp_names_map = None
+        if self.hierarchy:
+            several_transforms = len(transforms) > 1
+            partial_transform = (
+                "components" in transforms[0]
+                and transforms[0]["components"] != self.components
```

Comment:
It will raise an error, since we're comparing a list with a pandas Index.
Also this comparison is not order-agnostic. Can we improve it?